File: MP.Web/ClientApp/src/app/entities/artist.ts
import { Subject } from './subject';
import { Person } from './person';
import { Song } from './song';

export interface Artist extends Subject {
  name: string;
  yearStarted: number;
  yearQuit: number;

  text: string;

  currentMembers: Person[];
  pastMembers: Person[];
  songs: Song[];
}

File: MintPlayer.Dtos/Dtos/Artist.cs
using Nest;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace MintPlayer.Dtos.Dtos
{
	public class Artist : Subject
	{
		public string Name { get; set; }
		public int? YearStarted { get; set; }
		public int? YearQuit { get; set; }

		public List<Person> PastMembers { get; set; }
		public List<Person> CurrentMembers { get; set; }
		public List<Song> Songs { get; set; }

		[JsonIgnore]
		public CompletionField NameSuggest => new CompletionField { Input = new[] { Name } };
	}
}
